src/data/equiv/algebra.lean

```diff
@@ -505,12 +510,26 @@ variables [semiring α] [semiring β]
 def to_ring_hom (e : α ≃+* β) : α →+* β :=
 { .. e.to_mul_equiv.to_monoid_hom, .. e.to_add_equiv.to_add_monoid_hom }
 
+/-- Reinterpret a ring equivalence as a monoid homomorphism. -/
+def to_monoid_hom (e : α ≃+* β) : α →* β := e.to_ring_hom.to_monoid_hom
+
+/-- Reinterpret a ring equivalence as an add_monoid homomorphism. -/
+def to_add_monoid_hom (e : α ≃+* β) : α →+ β := e.to_ring_hom.to_add_monoid_hom
+
 /-- Interpret an equivalence `f : α ≃ β` as a ring equivalence `α ≃+* β`. -/
 def of (e : α ≃ β) [is_semiring_hom e] : α ≃+* β :=
 { .. e, .. monoid_hom.of e, .. add_monoid_hom.of e }
 
 instance (e : α ≃+* β) : is_semiring_hom e := e.to_ring_hom.is_semiring_hom
 
+def is_ring_hom_of_mul_equiv {R : Type*} {S : Type*} [ring R] [ring S]
+  (h : R ≃* S) (H: ∀ x y : R, h (x + y) = h x + h y) : is_ring_hom h :=
+@ring_hom.is_ring_hom _ _ _ _ $ ring_hom.mk' h.to_monoid_hom H
+
+def of_mul_equiv {R : Type*} {S : Type*} [ring R] [ring S] (h : R ≃* S)
```

Comment:
Possibly rename to `mul_equiv.to_semiring_hom`, so we can write `h.to_semiring_hom H`? Not certain this is better.